pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -482,15 +483,15 @@ func (c *InstallerController) ensureInstallerPod(nodeName string, operatorSpec *
 		"-v=4", // TODO: Make this configurable?
 		fmt.Sprintf("--revision=%d", revision),
 		fmt.Sprintf("--namespace=%s", pod.Namespace),
-		fmt.Sprintf("--pod=%s", c.configMaps[0]),
+		fmt.Sprintf("--pod=%s", c.configMaps[0].Name),
 		fmt.Sprintf("--resource-dir=%s", hostResourceDirDir),
 		fmt.Sprintf("--pod-manifest-dir=%s", hostPodManifestDir),
 	}
-	for _, name := range c.configMaps {
-		args = append(args, fmt.Sprintf("--configmaps=%s", name))
+	for _, cm := range c.configMaps {
+		args = append(args, fmt.Sprintf("--configmaps=%s", cm.Name))
```

Comment:
You're losing information here.  The fact that your signature changed above suggests that the information needs to preserved all the way to the command this is running.  In this case, `installerpod/cmd.go` needs to know that some items can 404 or you'll just permafail the cluster.

Reply:
Also, add a unit test to that command so you can see it work.